dali/base/dafdesc.cpp

```diff
@@ -1353,14 +1357,29 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
                 *_interleaved = interleaved;
         }
 
-        const char * historyStr = "History";
-        if (mb.remaining() > strlen(historyStr))
+        // If we have more serialised data in MemoryBuffer
+        if (mb.remaining() >= sizeof(unsigned short))
         {
-            const char * buffer = mb.toByteArray()+mb.getPos();
-            if (nullptr != strstr(buffer, historyStr))
-                history.setown(createPTree(mb));
+            unsigned short extendedVersion;
+            mb.read(extendedVersion);
+
+            if (extendedVersion >= VERSION_MINHISTORY)
+            {
+                //If extendedVersion >= 1 then we should have stored History
+                const char * historyStr = "History";
+                if (mb.remaining() > strlen(historyStr))
+                {
+                    const char * buffer = mb.toByteArray()+mb.getPos();
+                    if (nullptr != strstr(buffer, historyStr))
+                        history.setown(createPTree(mb));
+                    else
+                        history.setown(createPTree(historyStr));
```

Comment:
Your suggestion is wrong. That code fragment doesn't deserialise the history, but create a new and empty one. 

My code deseirialise history if there is a valid desriailisable one (contains valid characters to create xpath)  or create a new and empty one if not.
And what happened if the memory buffer destroyed/altered (after it serialised, trough the transport, etc) somehow (e.g. overwritten, truncated)? I know because I experienced it. Crash/exception somewhere other place (in the tree creation). Should we continue to process if we (can) know there is a missing bit and it will be unsuccessful?

So, I can throw an exception if there is not the expected (vital) string in in the mb from current position, like this:
```
const char * buffer = mb.toByteArray()+mb.getPos();
 if (nullptr != strstr(buffer, historyStr))
     history.setown(createPTree(mb));    // There is a valid History in mb deserialise it
else
    throw MakeStringException(-1,"Missing History from the extended version of FileDescriptor!");
```
or I try to catch it with:
```
try
     history.setown(createPTree(mb));    // Try to deserialise the content of the mb
catch(...)
...
    throw MakeStringException(-1,"Missing History from the extended version of FileDescriptor!");
```
Either way I think I should handle/prepare for the problem (a necessary paranoid behaviour to survive).
However I will implement what you want, with or without checking, because without your acceptance this code won't be merged.
In this case the code would be this:
```
 if (extendedVersion >= VERSION_MINHISTORY)
    history.setown(createPTree(mb));    // Deserialise the content of the mb
    // Further deserialisations should place here
 else
   throw MakeStringException(-1,"FileDescriptor extended serialization version shouldn't be zero!");
``